/tmp/brut_util_Jar_xxxxxxx.tmp conflict when build in multi-thread

### Information

1. **Apktool Version (`2.4.0`)** -
2. **Operating System (Linux)** -
3. **APK From? (Other)** -

### Stacktrace/Logcat

```
brut.androlib.AndrolibException: brut.androlib.AndrolibException: brut.common.BrutException: could not exec: [/tmp/brut_util_Jar_2231889210591345809.tmp, compile, --dir, /home/jenkins/data/jenkins/workspace/test-android_single_master-channels/working/channels/huawei/res, --legacy, -o, /home/jenkins/data/jenkins/workspace/test-android_single_master-channels/working/channels/huawei/build/resources.zip]
	at brut.androlib.Androlib.buildResourcesFull(Androlib.java:503)
	at brut.androlib.Androlib.buildResources(Androlib.java:426)
	at brut.androlib.Androlib.build(Androlib.java:305)
	at brut.androlib.Androlib.build(Androlib.java:269)
	at cn.mucang.tools.channel.Main.build(Main.java:221)
	at cn.mucang.tools.channel.Main.buildApkPerChannel(Main.java:192)
	at cn.mucang.tools.channel.Main.access$000(Main.java:34)
	at cn.mucang.tools.channel.Main$1.run(Main.java:105)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: brut.androlib.AndrolibException: brut.common.BrutException: could not exec: [/tmp/brut_util_Jar_2231889210591345809.tmp, compile, --dir, /home/jenkins/data/jenkins/workspace/test-android_single_master-channels/working/channels/huawei/res, --legacy, -o, /home/jenkins/data/jenkins/workspace/test-android_single_master-channels/working/channels/huawei/build/resources.zip]
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:362)
	at brut.androlib.res.AndrolibResources.aaptPackage(AndrolibResources.java:601)
	at brut.androlib.Androlib.buildResourcesFull(Androlib.java:481)
	... 10 more
Caused by: brut.common.BrutException: could not exec: [/tmp/brut_util_Jar_2231889210591345809.tmp, compile, --dir, /home/jenkins/data/jenkins/workspace/test-android_single_master-channels/working/channels/huawei/res, --legacy, -o, /home/jenkins/data/jenkins/workspace/test-android_single_master-channels/working/channels/huawei/build/resources.zip]
	at brut.util.OS.exec(OS.java:100)
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:358)
	... 12 more
Caused by: java.io.IOException: Cannot run program "/tmp/brut_util_Jar_2231889210591345809.tmp": error=26, 文本文件忙
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at brut.util.OS.exec(OS.java:93)
	... 13 more
Caused by: java.io.IOException: error=26, 文本文件忙
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 14 more
```

### Steps to Reproduce
1. Use ExecutorService execute `new Androidlib(apkOptions).build(dir, unsignedApkFile)` in multi-thread.

2. sometimes it will throw BrutException.  

3. I trace code and find some problem, I think it maybe caused the exception.

   ```java
   // Jar.java
   public static File extractToTmp(String resourcePath, String tmpPrefix, Class clazz) throws BrutException {
           try {
               InputStream in = clazz.getResourceAsStream(resourcePath);
               if (in == null) {
                   throw new FileNotFoundException(resourcePath);
               }
               File fileOut = File.createTempFile(tmpPrefix, null);
               fileOut.deleteOnExit();
               OutputStream out = new FileOutputStream(fileOut);
               IOUtils.copy(in, out);
               in.close();
               out.close();
               return fileOut;
           } catch (IOException ex) {
               throw new BrutException("Could not extract resource: " + resourcePath, ex);
           }
       }
   
   // File.java
   private static class TempDirectory {
           private TempDirectory() { }
   
           // temporary directory location
           private static final File tmpdir = new File(AccessController
               .doPrivileged(new GetPropertyAction("java.io.tmpdir")));
           static File location() {
               return tmpdir;
           }
   
           // file name generation
           private static final SecureRandom random = new SecureRandom();
           static File generateFile(String prefix, String suffix, File dir)
               throws IOException
           {
               long n = random.nextLong();
               if (n == Long.MIN_VALUE) {
                   n = 0;      // corner case
               } else {
                   n = Math.abs(n);
               }
   
               // Use only the file name from the supplied prefix
               prefix = (new File(prefix)).getName();
   
               String name = prefix + Long.toString(n) + suffix;
               File f = new File(dir, name);
               if (!name.equals(f.getName()) || f.isInvalid()) {
                   if (System.getSecurityManager() != null)
                       throw new IOException("Unable to create temporary file");
                   else
                       throw new IOException("Unable to create temporary file, " + f);
               }
               return f;
           }
       }
   ```

   I think that random.nextLong()  return same values sometimes.

   

### Frameworks
If this APK is from an OEM ROM (Samsung, HTC, LG). Please attach framework files
(`.apks` that live in `/system/framework` or `/system/priv-app`)

### APK
If this APK can be freely shared, please upload/attach a link to it.

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything?
2. If you are trying to install a modified apk, did you resign it?
3. Are you using the latest apktool version?

--- a/brut.j.util/src/main/java/brut/util/Jar.java
+++ b/brut.j.util/src/main/java/brut/util/Jar.java
@@ -16,15 +16,21 @@
  */
 package brut.util;
 
-import brut.common.BrutException;
+import org.apache.commons.io.IOUtils;
 
-import java.io.*;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
 
-import org.apache.commons.io.IOUtils;
+import brut.common.BrutException;
 
 /**
  * @author Ryszard Wiśniewski <brut.alll@gmail.com>
@@ -79,7 +85,14 @@ abstract public class Jar {
             if (in == null) {
                 throw new FileNotFoundException(resourcePath);
             }
-            File fileOut = File.createTempFile(tmpPrefix, null);
+            long n = ThreadLocalRandom.current().nextLong();
+            if (n == Long.MIN_VALUE) {
+                n = 0;      // corner case
+            } else {
+                n = Math.abs(n);
+            }
+            String suffix = Long.toString(n) + ".tmp";
+            File fileOut = File.createTempFile(tmpPrefix, suffix);
             fileOut.deleteOnExit();
             OutputStream out = new FileOutputStream(fileOut);
             IOUtils.copy(in, out);

